Src/VimCore/MefComponents.fs

```diff
@@ -72,7 +72,9 @@ type internal TrackingLineColumn
     /// Update the internal tracking information based on the new ITextSnapshot
     member x.OnBufferChanged (e: TextContentChangedEventArgs) =
         match _line with 
-        | Some snapshotLine -> x.AdjustForChange snapshotLine e
+        | Some snapshotLine ->
+            if e.AfterVersion <> snapshotLine.Snapshot.Version then

```

Comment:
Is this optimizing for the case of avoiding calculations on an undo / redo?

Reply:
When marks are set *during* the `OnBufferChanged` event, they will be already be correct for the current snapshot. When there are multiple listeners to that event, we don't know the order in which they will be called. So it can easily happen (and I saw it during debugging) that an listener who is called early and creates a mark (like the `TextChangeTracker`) can set a mark and then a later listener (like the `TrackingLineColumn`) can try to track that very mark.

So it's not just an optimization because if the snapshot line is already up-to-date with the after version we would corrupt the position of the mark by trying to adjust it.